### PR TITLE
feat: optimize collectible list in pfp

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes mf-4063

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/shared/src/UI/components/AssetsManagement/CollectionList.tsx [hunk 1]
```diff
@@ -150,6 +151,7 @@ export interface CollectionListProps
     gridProps?: CollectibleGridProps
     disableSidebar?: boolean
     disableWindowScroll?: boolean
+    selectedAsset?: Web3Helper.NonFungibleAssetScope
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/AssetsManagement/CollectionList.tsx]: Please set S and T generic parameters.

File: packages/shared/src/UI/components/AssetsManagement/CollectibleCard.tsx [hunk 2]
```diff
@@ -49,6 +58,19 @@ export interface CollectibleCardProps extends HTMLProps<HTMLDivElement> {
     disableNetworkIcon?: boolean
     /** disable inspect NFT details */
     disableInspect?: boolean
+    selectedAsset?: Web3Helper.NonFungibleAssetScope
+}
+
+export function isSameNFT(
+    pluginID: NetworkPluginID,
+    a: Web3Helper.NonFungibleAssetScope,
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/AssetsManagement/CollectibleCard.tsx]:
```suggestion
    a: Web3Helper.NonFungibleAssetAll,
```

File: packages/shared/src/UI/components/AssetsManagement/CollectionList.tsx [hunk 3]
```diff
@@ -150,6 +151,7 @@ export interface CollectionListProps
     gridProps?: CollectibleGridProps
     disableSidebar?: boolean
     disableWindowScroll?: boolean
+    selectedAsset?: Web3Helper.NonFungibleAssetScope<S, T>
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/AssetsManagement/CollectionList.tsx]:
```suggestion
    selectedAsset?: Web3Helper.NonFungibleAssetAll
```

File: packages/shared/src/UI/components/AssetsManagement/CollectibleCard.tsx [hunk 4]
```diff
@@ -49,6 +58,19 @@ export interface CollectibleCardProps extends HTMLProps<HTMLDivElement> {
     disableNetworkIcon?: boolean
     /** disable inspect NFT details */
     disableInspect?: boolean
+    selectedAsset?: Web3Helper.NonFungibleAssetScope
```
Comment (UncleBill) [inline packages/shared/src/UI/components/AssetsManagement/CollectibleCard.tsx]: Should not passing `selectedAsset` to this component but an `isSelected` status instead.

File: packages/shared/src/UI/components/AssetsManagement/CollectibleCard.tsx [hunk 5]
```diff
@@ -49,6 +58,19 @@ export interface CollectibleCardProps extends HTMLProps<HTMLDivElement> {
     disableNetworkIcon?: boolean
     /** disable inspect NFT details */
     disableInspect?: boolean
+    selectedAsset?: Web3Helper.NonFungibleAssetAll
```
Comment (UncleBill) [inline packages/shared/src/UI/components/AssetsManagement/CollectibleCard.tsx]: `CollectibleCard` should not reply on `selectedAsset`, do not pass `selectedAsset` to this component but an `isSelected` (or `selected`) status instead.

File: packages/shared/src/UI/components/AssetsManagement/CollectibleCard.tsx [hunk 6]
```diff
@@ -41,6 +43,13 @@ const useStyles = makeStyles()((theme) => ({
         width: '100%',
         height: '100%',
     },
+    icon: {
```
Comment (UncleBill) [inline packages/shared/src/UI/components/AssetsManagement/CollectibleCard.tsx]: `indicator`, it's a selected indicator.

**Review by guanbinrui — APPROVED** [review]

**Review by UncleBill — APPROVED** [review]

**Pull request merged**